test/ecdsa_adaptor.json

```diff
@@ -0,0 +1,56 @@
+[
+    {
+        "adaptor_sig": "036c2f34833942e6eebb9e01b71c1d5f6ea185e7af607805ab5a39a829ad7a30400220f57f812e043e0f5b5c8c22a5c6efe6f76cdaa71490f7b3a3adb7a39101bf7ad4ee4d049e573adee6647ab7d97445d87c4252d91a627935269e99703a7d4161a5b4e2675d09e0da65deb2ed6210e5e99244153e9600ed17d90f6f31853e2c5ff8680adcbd1393d1ae8eed5cd8a8e396275be95506bf40a20ac22f66ca875bf4",
+        "message_hash": "8131e6f4b45754f2c90bd06688ceeabc0c45055460729928b4eecf11026a9e2d",
+        "public_signing_key": "039b8327d929a0e45285c04d19c9fffbee065c266b701972922d807228120e43f3",
```

Comment:
It would be nice to have the private key for the `public_signing_key`s so that `ecdsa_adaptor_encrypt` can be tested with these vectors.

Reply:
The problem is that I don't want to specify exactly how to do nonce generation (only what must be input into the nonce generation function). So what you get out from `ecdsa_adaptor_encrypt` is not going to be the same on all implementations since the spec allows nonce generation to be done in different ways

For example what I do in my own implementation is to hash all the things as described above, add randomness into the hash and then to use the output to seed ChaCha20 from which I generate all nonces. The reason I do this is that in my library you can compose arbitrary sigma ZKPs which may need hundreds of nonces. This would be an awkward way to specify this particular proof though since you only need one nonce for the proof.

Reply:
That makes sense. What if you included the nonce in the test vector? That way `ecdsa_adaptor_encrypt` could be tested with the private key and nonce from the test. Albeit, it would have to take a nonce as an optional input, which slightly changes the function signature from the spec.

Reply:
I think you're right. We could provide some separate signing vectors as well as a test to check decryption on its own,

Reply:
On second thought I think this is not worth it on its own -- if your signing passes the verification then it will work. The only security concern then is nonce generation which as I want to leave out of scope for this scheme.